atom/browser/browser_mac.mm

```diff
@@ -87,6 +87,23 @@
 void Browser::SetAppUserModelID(const base::string16& name) {
 }
 
+void Browser::SetUserActivity(const std::string& type, const std::map<std::string, std::string>& user_info) {
+  NSString* type_ns = [NSString stringWithUTF8String:type.c_str()];
+  NSUserActivity *user_activity = [[NSUserActivity alloc] initWithActivityType:type_ns];
+
+  NSMutableArray* user_info_args = [[NSMutableArray alloc] init];
+  for (auto const &pair : user_info) {
+    NSString* key_ns = [NSString stringWithUTF8String:pair.first.c_str()];
+    NSString* value_ns = [NSString stringWithUTF8String:pair.second.c_str()];
+    
+    [user_info_args addObject:key_ns];
+    [user_info_args addObject:value_ns];
```

Comment:
I believe the order is value,key,value,key,value,key etc https://developer.apple.com/library/mac/documentation/Cocoa/Reference/Foundation/Classes/NSDictionary_Class/index.html#//apple_ref/occ/instm/NSDictionary/initWithObjectsAndKeys: